### PR TITLE
planner,tests: migrate part of planner tests to mysql_tester

### LINKED ISSUE #47199: Move IT in the sub-packages of `planner/` to explaintest,
This PR is part of https://github.com/pingcap/tidb/issues/47173. The `planner` package is too huge. I'll finish them in two issues/prs: one for some misc packages, one for the main packages (`/planner/core`)

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Issue Number: close #47199

Problem Summary:

As stated in https://github.com/pingcap/tidb/issues/45961, I'll migrate most of the tests using "CreateMockStore"/testkit to `mysql_tester`, as it can reduce the consumption of recreating TiDB environment for every tests.

### What is changed and how it works?

Auto generate the test, and manually verify the codes to check whether they are proper to test in `mysql_tester`. I also removed some data from the `json` test data, but some format may changed (as my script to remove the data according to the removed tests will automatically format the json... ).

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/cardinality/selectivity_test.go [hunk 1]
```diff
@@ -724,34 +535,6 @@ func TestDNFCondSelectivity(t *testing.T) {
...
 
-func TestIndexEstimationCrossValidate(t *testing.T) {
-	store := testkit.CreateMockStore(t)
-	tk := testkit.NewTestKit(t, store)
-	tk.MustExec("use test")
-	tk.MustExec("drop table if exists t")
-	tk.MustExec("create table t(a int, b int, key(a,b))")
-	tk.MustExec("insert into t values(1, 1), (1, 2), (1, 3), (2, 2)")
-	tk.MustExec("analyze table t")
-	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/statistics/table/mockQueryBytesMaxUint64", `return(100000)`))
```
Comment (Defined2014) [inline planner/cardinality/selectivity_test.go:735]: How is failpoint handled?

File: planner/core/casetest/testdata/expression_rewriter_suite_in.json [hunk 2]
```diff
@@ -1,17 +1 @@
...
-    ]
-  },
-  {
-    "name": "TestBitFuncsReturnType",
-    "cases": [
-      "select bit_or(a * b) from t"
-    ]
-  }
-]
+[]
```
Comment (Defined2014) [inline planner/core/casetest/testdata/expression_rewriter_suite_in.json:1]: Still need this file or not?

File: planner/core/casetest/testdata/expression_rewriter_suite_out.json [hunk 3]
```diff
@@ -1,80 +1 @@
...
-          "StreamAgg 1.00 root  funcs:bit_or(Column#6)->Column#4",
-          "└─TableReader 1.00 root  data:StreamAgg",
-          "  └─StreamAgg 1.00 cop[tikv]  funcs:bit_or(cast(mul(cast(test.t.a, double BINARY), cast(test.t.b, double BINARY)), bigint(20) BINARY))->Column#6",
-          "    └─TableFullScan 1.00 cop[tikv] table:t keep order:false"
-        ]
-      }
-    ]
-  }
-]
+[]
```
Comment (Defined2014) [inline planner/core/casetest/testdata/expression_rewriter_suite_out.json:1]: ditto

File: planner/core/casetest/testdata/integration_serial_suite_out.json
Comment (Defined2014) [inline planner/core/casetest/testdata/integration_serial_suite_out.json:1]: ditto

File: planner/core/casetest/testdata/predicate_simplification_in.json [hunk 4]
```diff
@@ -1,35 +1 @@
...
-      "select 1 from t A, t B where A.f <> 3 and B.f in (1,2,3) and A.f <> 1 and A.f <> 2 -- on different columns. No simplification should be done.",
-      "select 1 from t A, t B where B.f <> 2 and A.f <> 3 and B.f in (1,2,3) and A.f in (3,1,4) and A.f <> 1 and A.f <> 2 -- simplification for two columns.",
-      "select f from ts use index() where f <> '1' and f in ('1','2','3') -- Simple case with string type",
-      "select count(*) cnt from ts where f <> '1' and f in ('1','2','3') group by a having cnt > 100  -- aggregate  ",
-      "select f from t where f <> NULL  and f in (1,2,3) -- Special case of NULL with no simplification.",
-      "select f from t where f != NULL  and f in (NULL,2,3) -- Special case of NULL with no simplification."
-    ]
-  }
-]
+[]
```
Comment (Defined2014) [inline planner/core/casetest/testdata/predicate_simplification_in.json:1]: ditto

File: planner/core/casetest/testdata/predicate_simplification_out.json [hunk 5]
```diff
@@ -1,207 +1 @@
...
-        "Plan": [
-          "TableReader 0.00 root  data:Selection",
-          "└─Selection 0.00 cop[tikv]  in(test.t.f, NULL, 2, 3), ne(test.t.f, NULL)",
-          "  └─TableFullScan 10000.00 cop[tikv] table:t keep order:false, stats:pseudo"
-        ]
-      }
-    ]
-  }
-]
+[]
```
Comment (Defined2014) [inline planner/core/casetest/testdata/predicate_simplification_out.json:1]: ditto

File: planner/core/casetest/testdata/expression_rewriter_suite_in.json (hunk 2 above)
Comment (YangKeao) [reply planner/core/casetest/testdata/expression_rewriter_suite_in.json]: I'll remove all empty json files (and corresponding loading functions).

File: planner/cardinality/selectivity_test.go (hunk 1 above)
Comment (YangKeao) [reply planner/cardinality/selectivity_test.go]: This test (`TestIndexEstimationCrossValidate`) is actually not moved. I have checked with diff that I didn't move any test containing a failpoint 🤔 .

**Comment by winoros** [comment]
Seems that the code coverage will not be affected?

**Comment by YangKeao** [comment]
> Seems that the code coverage will not be affected?

Yes. The code coverage rate shouldn't be affected. Let me investigate the report 🤔

**Review by hawkingrei — APPROVED** [review]

**Comment by Defined2014** [comment]
> > Seems that the code coverage will not be affected?
> 
> `codecov/project/unit` decreased, which is expected (if it only gives the coverage by unit tests). I don't know whether the coverage of `tests/integrationtest` is included, and if it is, which component it's included in.
> 
> @Defined2014 Do you know more about the code coverage report?

I think `tests/integrationtest` will be include in code coverage with flag `integration`. I added it before, see https://github.com/pingcap/tidb/pull/45930 and https://github.com/PingCAP-QE/ci/pull/2342

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**